toolkit/quickrebuild_autoconfig.mk

```
@@ -0,0 +1,80 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Contains:
+#	- build optimization flag auto configuration
+
+######## QUICK_REBUILD AUTO CONFIGURE ########
+
+# The QUICK_REBUILD* flags are special flags that will try to build the toolchain and packages as quickly as possible. They will
+# automatically set REBUILD_TOOLS, USE_CCACHE, REBUILD_TOOLCHAIN, DELTA_BUILD, INCREMENTAL_TOOLCHAIN, and ALLOW_TOOLCHAIN_DOWNLOAD_FAIL to 'y'.
+QUICK_REBUILD           ?= n
+QUICK_REBUILD_TOOLCHAIN ?= n
+QUICK_REBUILD_PACKAGES  ?= n
+
+ifeq ($(QUICK_REBUILD),y)
+QUICK_REBUILD_TOOLCHAIN = y
+QUICK_REBUILD_PACKAGES  = y
+endif
+
+######## QUICK_REBUILD TOOLCHAIN ########
+
+ifeq ($(QUICK_REBUILD_TOOLCHAIN),y)
+# If any of these are already set to 'n', report an error. The whole point of using QUICK_REBUILD is
+# to enable these. Both INCREMENTAL_TOOLCHAIN and DELTA_BUILD are checked because DELTA_BUILD implies
+# the depcrecated INCREMENTAL_TOOLCHAIN.
+ifneq ($(filter n,$(REBUILD_TOOLCHAIN)),)
+$(error REBUILD_TOOLCHAIN cannot be used with REBUILD_TOOLCHAIN explicitly set to 'n')
+endif
+ifneq ($(filter n,$(INCREMENTAL_TOOLCHAIN)),)
+$(error REBUILD_TOOLCHAIN cannot be used with INCREMENTAL_TOOLCHAIN explicitly set to 'n')
+endif
+ifneq ($(filter n,$(DELTA_BUILD)),)
+$(error REBUILD_TOOLCHAIN cannot be used with DELTA_BUILD explicitly  set to 'n')
+endif
+ifneq ($(filter n,$(ALLOW_TOOLCHAIN_DOWNLOAD_FAIL)),)
+$(error REBUILD_TOOLCHAIN cannot be used with ALLOW_TOOLCHAIN_DOWNLOAD_FAIL explicitly set to 'n')
+endif
```

Shouldn't the error messages in this section refer to `QUICK_REBUILD_TOOLCHAIN`? Similar to what we're printing for `QUICK_REBUILD_PACKAGES` below?

```suggestion
ifneq ($(filter n,$(REBUILD_TOOLCHAIN)),)
$(error QUICK_REBUILD_TOOLCHAIN cannot be used with REBUILD_TOOLCHAIN explicitly set to 'n')
endif
ifneq ($(filter n,$(INCREMENTAL_TOOLCHAIN)),)
$(error QUICK_REBUILD_TOOLCHAIN cannot be used with INCREMENTAL_TOOLCHAIN explicitly set to 'n')
endif
ifneq ($(filter n,$(DELTA_BUILD)),)
$(error QUICK_REBUILD_TOOLCHAIN cannot be used with DELTA_BUILD explicitly  set to 'n')
endif
ifneq ($(filter n,$(ALLOW_TOOLCHAIN_DOWNLOAD_FAIL)),)
$(error QUICK_REBUILD_TOOLCHAIN cannot be used with ALLOW_TOOLCHAIN_DOWNLOAD_FAIL explicitly set to 'n')
endif
```